master/internal/api_experiment.go

```diff
@@ -580,6 +582,35 @@ func (a *apiServer) PatchExperiment(
 		}
 	}
 
+	if req.Experiment.MaxSlots != nil || req.Experiment.Weight != nil ||
+		req.Experiment.Priority != nil {
+		dbExp, err := a.m.db.ExperimentByID(int(req.Experiment.Id))
+		resources := dbExp.Config.Resources()
+		if err != nil {
+			return nil, errors.Wrapf(err,
+				"error fetching experiment config from database: %d", req.Experiment.Id)
+		}
+		if req.Experiment.MaxSlots != nil {
+			resources.SetMaxSlots(ptrs.IntPtr(int(req.Experiment.MaxSlots.Value)))
+			a.m.system.TellAt(actor.Addr("experiments", req.Experiment.Id),
+				sproto.SetGroupMaxSlots{MaxSlots: ptrs.IntPtr(int(req.Experiment.MaxSlots.Value))})
+		}
+		if req.Experiment.Weight != nil {
+			resources.SetWeight(float64(req.Experiment.Weight.Value))
+			a.m.system.TellAt(actor.Addr("experiments", req.Experiment.Id),
+				job.SetGroupWeight{Weight: float64(req.Experiment.Weight.Value)})
+		}
+		if req.Experiment.Priority != nil {
+			resources.SetPriority(ptrs.IntPtr(int(req.Experiment.Priority.Value)))
+			a.m.system.TellAt(actor.Addr("experiments", req.Experiment.Id),
```

Comment:
can we make a helper func to get this actor.Addr and try to make everything use it? it would be really useful when if we try to refactor to not accidentally break stuff.